examples/nextjs-shadcn/src/app/editor-example.tsx

```diff
@@ -162,11 +168,54 @@ export default function EditorExample() {
     };
   }, [checksummedAddress]);
 
+  React.useEffect(() => {
+    setInitialModActionAndContext(null);
+  }, [currentMod]);
+
   return (
     <form onSubmit={handleSubmit}>
       <div className="p-2 border border-input rounded-md">
         <EditorContent
           editor={editor}
+          onPaste={async (e) => {
+            const isPastingText = e.clipboardData.getData("text");
+            if (isPastingText) return;
+
+            const files = e.clipboardData.files;
+
+            // Mod expects a blob, so we convert the file to a blob
+            const imageFiles = await Promise.all(
+              Array.from(files)
+                .filter(({ type }) => {
+                  return type.startsWith("image/");
+                })
+                .map(async (file) => {
+                  const arrayBuffer = await file.arrayBuffer();
+                  const blob = new Blob([new Uint8Array(arrayBuffer)], {
+                    type: file.type,
+                  });
+                  return { blob, ...file };
+                })
+            );
+
+            if (imageFiles.length === 0) return;
+
+            // Open the imgur upload mod
+            setCurrentMod(
+              creationModsExperimental.find(
+                (mod) => mod.slug === "imgur-upload"
+              )
+            );
+
+            setInitialModActionAndContext({
```

Comment:
I'm not a fan of this

A) I don't like the name `initialModActionAndContext`
B) I don't like that it's decoupled from the currentMod. If you change the currentMod, it won't automatically reset the `initialModActionAndContext` and lead to bugs or inconsistencies
C) It adds fragility in coupling the implementation details of a version of a Mod to end users codebases, and significantly expands the API complexity and surface area

Reply:
think we need to go back to the drawing board on this one - I thought we'd discussed mods publishing transforms or handlers or a better name (inputs they accept) and then using those to trigger mods from paste?

I think slash commands are a good benchmark for this. If Mod were to support `/chatgpt imagine [Prompt input type text]` then any Mod that supports an image file has the singular argument could automatically also be a paste handler

Reply:
so a better abstraction is an api for mod commands that also supports what we need for pastes

Reply:
Thanks for the feedback!

>I thought we'd discussed mods publishing transforms or handlers or a better name (inputs they accept) and then using those to trigger mods from paste?

Don't recall it being discussed but happy to go that route. Mods specifying entrypoints to handle different input types is a natural step in composability as well, so would be a good addition to the API